### PR TITLE
E2E Test Environment Vars

### PR DESCRIPTION
**Pull request by aarongranick-okta**
centralizes use of environment variables in E2E tests. Used by:
- grunt
- protractor
- angular app
- react app

Replaces hardcoded values with new vars & "defaults": WIDGET_CLIENT_ID, and user names.

Adds a basic static web server for testing the "basic" specs

Adds README/instructions for manual E2E testing

Upgrades angular test app (to v.7) and receives all params from ENV vars (remove pretest step)
Some fixes to angular app & use latest version of OIDC.

### CODE REVIEW

File: test/e2e/README.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+```
+
+You should only need to do this once.
+
+## Run specs
+
+To see protractor options simply run:
+
+```
+npx protractor
```
Comment (haishengwu-okta) [inline test/e2e/README.md]: curious why suggest to use `npx` over `yarn`?

File: test/e2e/angular-app/extra-webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+const webpack = require('webpack');
+const env = {};
+
+// List of environment variables made available to the app
+[
+  'PORT',
+  'WIDGET_TEST_SERVER',
+  'WIDGET_CLIENT_ID'
+].forEach(function (key) {
+  env[key] = JSON.stringify(process.env[key]);
```
Comment (haishengwu-okta) [inline test/e2e/angular-app/extra-webpack.config.js]: why need `stringify`?

File: test/e2e/angular-app/tslint.json [hunk 3]
```diff
@@ -18,7 +18,6 @@
     "forin": true,
     "import-blacklist": [
       true,
-      "rxjs",
```
Comment (haishengwu-okta) [inline test/e2e/angular-app/tslint.json]: curious why this got removed but not the other?

File: test/e2e/README.md (hunk 1 above)
Comment (aarongranick-okta) [reply test/e2e/README.md]: This is literally running ./node_modules/.bin/protractor

`yarn protractor` would run the "protractor" script in package.jsonm

**Comment by aarongranick-okta** [comment]
> curious why angular-app and react-app need `dotenv` dependency where variables are pass to those app via process.env?

All the env handling is centralized in test/e2e/env.js  This file uses dotenv and a set of defaults to build a consistent set of environment variables for each process.

React & Angular apps run in their own process and load this "env.js" file to get the same set of environment variables as grunt, etc. (even if none are passed on CLI, which is preferred)

File: test/e2e/angular-app/tslint.json (hunk 3 above)
Comment (aarongranick-okta) [reply test/e2e/angular-app/tslint.json]: This file was auto-modified by `ng upgrade`

File: test/e2e/angular-app/extra-webpack.config.js (hunk 2 above)
Comment (aarongranick-okta) [reply test/e2e/angular-app/extra-webpack.config.js]: If the value is a string it must be "quoted" or we will get syntax error when webpack substitutes it in the source.

**Pull request merged**